### ISSUE TITLE
Requests by external library from an iframe open new tabs/windows

### ISSUE THREAD

**Issue by Eiskis** [jiahaog/nativefier#793]
### Details

- Are you nativefying a *public* website?: yes
- Feature request? Nope
- Full nativefier command used to build your app: `nativefier "https://www.freshworks.com/live-chat-software/"`
- Version of Nativefier (run `nativefier --version`): `v7.6.12`
- Version of node.js (run `node --version`): `v10.15.3`
- OS: `macOS Mojave`

### Explanation

First of all, I'm not 100 % sure of the exact issue here, but I have a straightforward replicable case. This ticket is not about Freshchat specifically, I think there's a more general root cause that could be addressed.

I have a web site that uses Freshchat, a support chat widget that can be embedded on web sites and apps. It works by loading a JS widget from their server. You can see details on [their docs](https://developers.freshchat.com/). As far as I can tell, this script loads an iframe and some more assets using multiple requests.

You can try it yourself with the following build:

```sh
npx nativefier "https://www.freshworks.com/live-chat-software/"
```

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/1101002/56319890-198f7780-6163-11e9-80fb-4b4a927f7ec5.gif)

What happens is that the host website gets loaded (happens with any website with this library, this is just an example), but a new tab then opens in a browser with a blank page and the chat widget (I think this is the iframe content that would normally be injected into the page). The host page still has the iframe that the widget normally loads, but I think the requests that are in the iframe head open in a new tab.

I tried adding freshchat.com to the `internal-urls` config param:

```sh
npx nativefier "https://www.freshworks.com/live-chat-software/" --internal-urls ".*(google|freshchat)\.com.*"
```

The result is the following:

<img width="1278" alt="Screenshot 2019-04-17 at 22 40 00" src="https://user-images.githubusercontent.com/1101002/56319570-5a3ac100-6162-11e9-82c6-4294582f73df.png">

So basically the same thing, but instead of opening a browser tab, a new window opens instead. This leads me to think that the `internal-urls` config is working (since it changes the behavior), but doesn't really achieve the goal here (which is that the iframe behavior should match freshworks.com).

Due to the way the library authors provide the library, it's not really easy to use this from NPM and only load it up conditionally, and I do want to use the widget on my product, so I'm looking for a solution that would make this work. A lot of web services work in a similar way so I wonder how often this happens with other external widgets.

If anyone can point me in the right direction, would be happy to do more debugging, finding the solution and submitting a PR if needed.

**Comment by simonhyenes** [comment]
Same issue, freshchat tries to open a new window with internal-urls. Now I no longer include it in the internal-urls and it opens a new tab in my default browser. Meaning that the chat-bubble is not shown in the application.

**Comment by neongreen** [comment]
Has anybody found a workaround?

**Comment by neongreen** [comment]
(I'll be happy to fund finding a workaround for Freshchat specifically, if it helps.)

**Comment by wunderdojo** [comment]
Same issue here with Workast. They use the Freshchat widget which gets opened in a new tab or window depending on how you have --internal-urls set. Did a little digging by injecting a js file to log things. Freshchat is injected via a script which then inserts a div and iframe. Looks like the electron app treats the iframe as a second window. Since the injected js file isn't read until after the dom is completely loaded there's no opportunity to just remove the included script via something like:

`document.querySelector("script[src*='freshchat']").remove()`

If there was an option to inject a script that gets injected at the start of dom rendering that would be a workaround but ideally it would simply render iframes internally, not in a secondary window.
